promql/functions.go

```diff
@@ -170,8 +170,8 @@ func funcIrate(ev *evaluator, args Expressions) model.Value {
 
 // === sort(node model.ValVector) Vector ===
 func funcSort(ev *evaluator, args Expressions) model.Value {
-	byValueSorter := vectorByValueHeap(ev.evalVector(args[0]))
-	sort.Sort(byValueSorter)
+	byValueSorter := vectorByReverseValueHeap(ev.evalVector(args[0]))
```

Comment:
I must say I'm not too happy with the general approach. I generally expect `reverse ∘ reverse` to be a no-op. This is way to easy to be changed back by someone later on who thinks the same.

It's probably not worth investing much effort into a "pretty" solution. But let's have proper doc strings here and in `bottomk` that explain what we are doing and why.